statick_tool/discovery_plugin.py

```diff
@@ -36,6 +37,40 @@ def scan(
         use it to filter which files the plugin detects.
         """
 
+    def walk_once(self, package: Package) -> None:
+        """Walk the package path exactly once to discover files for analysis."""
+        if package.walked:
+            return
+
+        for root, _, files in os.walk(package.path):
+            for f in files:
+                full_path = os.path.join(root, f)
+                abs_path = os.path.abspath(full_path)
+                file_output = self.get_file_cmd_output(full_path)
+                file_dict = {
+                    "name": f.lower(),
+                    "path": abs_path,
+                    "file_cmd_out": file_output,
+                }
+                package.files[abs_path] = file_dict
+
+        package.walked = True
+
+    def get_file_cmd_output(self, full_path: str) -> str:
+        """Run the file command (if it exists) on the supplied path."""
+        if not self.file_command_exists():
+            return ""
+
+        try:
+            output = subprocess.check_output(
+                ["file", full_path], universal_newlines=True
+            )  # type: str
+            return output
+        except subprocess.CalledProcessError as ex:
```

Comment:
I'm not completely sure how this is being tested, but the coverage reports say it is. I also didn't find any unit tests specifically for this file. Do you have any suggestions on whether I should add a new unit test specifically for this file?